src/main/kotlin/racingcar/interfaces/ui/input/console/RacingCarInputConsole.kt

```diff
@@ -0,0 +1,31 @@
+package racingcar.interfaces.ui.input.console
+
+import racingcar.application.model.RaceInputCommand
+
+object RacingCarInputConsole {
+
+    fun input(): RaceInputCommand {
+        val carCount = queryNumber("자동차 대수는 몇 대인가요?")
+        val attemptCount = queryNumber("시도할 횟수는 몇 회인가요?")
+        return RaceInputCommand(carCount, attemptCount)
+    }
+
+    private fun queryNumber(question: String): Int =
+        try {
+            println(question)
+            val numberString = readln()
+            validateNumber(numberString)
+            numberString.toInt()
+        } catch (e: IllegalArgumentException) {
+            queryNumber(question)
+        }
+
+    private fun validateNumber(string: String?) {
+        try {
+            require(!string.isNullOrEmpty()) { "올바른 숫자를 입력해주세요" }
+            string.toInt()
+        } catch (e: NumberFormatException) {
+            throw IllegalArgumentException("올바른 숫자를 입력해주세요")
+        }
+    }
+}
```

Comment:
```suggestion

    private fun validateNumber(string: String) {
        try {
            require(string.isNotBlank()) { "올바른 숫자를 입력해주세요" }
            string.toInt()
        } catch (e: NumberFormatException) {
            throw IllegalArgumentException("올바른 숫자를 입력해주세요")
        }
    }
}
```

- `readln()` 을 통해 전달 받은 값이므로, 이미 null-safe 한거 같아요.
- kotlin 은 Not 을 포함한 검증문을 제공해주곤 합니다 😄 
- 음수도 모든 검증을 통과하고 핵심로직까지 전달되겠네요..!
  - 여기서 잠깐, '음수는 안된다'는 InputConsole 이 알아야 할 내용일까요?

Reply:
- `nullable`하게 받아서 문제가 생겼네요 ㅠㅠ [6755bc8](https://github.com/next-step/kotlin-racingcar/pull/808/commits/6755bc8a78ce29ba82840eda280de784d19fd983)
- 피드백주신 것 처럼 not을 활용하는 함수가 더 가독성이 좋은 거 같네요
- InputConsole이 받을 이유는 없어보입니다. 단순히 입력 받는 값의 타입(int, string)을 정해주고 전달하는 용도가 적합해보입니다! 대신 커맨드를 입력받고 전달하는 객체에 책임을 위임할지 비즈니스 로직을 처리하는 Race에서 처리할지 고민을 해보았습니다. 값 자체가 Race의 로직으로 들어오기에 적합하지 않은 값이기에 커맨드 객체를 생성하는 과정에서 검사를 하는 것이 좋다고 판단하여서 입력 모델 객체안에 로직을 추가했습니다! [8baecf4](https://github.com/next-step/kotlin-racingcar/pull/808/commits/8baecf4bd6fa4e11bc8d98f562d22481884f433e)

Reply:
- `입력 모델 객체안에 로직을 추가했습니다!` 👍 검증은 하면 할수록, 특히 view 에서 멀고 model 에서 가깝게 할수록 중복 검증이 줄어들고 핵심 로직도 안전해지니, 양측에서 모두 해주는 것도 좋을거 같습니다 😄 
- `RacingCarInputConsoleView` 외에 다른 Input 에서도 `RaceInputCoommand`사용 할 수 있으니, blank, null 등의 체크 로직 또한 `RaceInputCommand` 에게 넘기는 방법도 있을거 같아요!